Mlem/App/Views/Shared/Search/SearchView.swift

```diff
@@ -0,0 +1,155 @@
+                }
+            }
+            .onChange(of: selectedTab) {
+                Task { @MainActor in
+                    do {
+                        switch selectedTab {
+                        case .communities:
+                            if communities.isEmpty {
+                                communities = try await appState.firstApi.searchCommunities(query: query, page: 1, limit: 20)
+                            }
+                        case .users:
+                            if people.isEmpty {
+                                people = try await appState.firstApi.searchPeople(query: query, page: 1, limit: 20)
+                            }
+                        case .instances:
+                            if instances.isEmpty {
+                                instances = try await MlemStats.main.searchInstances(query: query)
+                            }
+                        }
+                    } catch {
+                        handleError(error)
+                    }
+                }
+            }
+    }
+    
+    @ViewBuilder
+    var content: some View {
+        FancyScrollView(scrollToTopTrigger: $resultsScrollToTopTrigger) { searchBarFocused = true } content: {
+            BubblePicker(
+                Tab.allCases, selected: $selectedTab,
+                withDividers: [.bottom],
+                label: { $0.rawValue.capitalized }
+            )
+            .padding(.top, -8)
+            LazyVStack(spacing: 0) {
+                switch selectedTab {
+                case .communities:
+                    SearchResultsView(results: communities) { community in
+                        CommunityListRow(community, readout: .subscribers)
+                    }
+                case .users:
+                    SearchResultsView(results: people) { person in
+                        PersonListRow(person, complications: [.instance, .date], readout: .postsAndComments)
+                    }
+                case .instances:
+                    SearchResultsView(results: instances) { instance in
+                        InstanceListRow(instance)
+                    }
+                }
+            }
+        }
+    }
+    
+    func refresh() async {
+        do {
+            if !query.isEmpty {
+                try await Task.sleep(for: .seconds(0.2))
+            }
+            communities.removeAll()
+            people.removeAll()
+            instances.removeAll()
+            switch selectedTab {
+            case .communities:
+                communities = try await appState.firstApi.searchCommunities(query: query, page: 1, limit: 20)
+            case .users:
+                people = try await appState.firstApi.searchPeople(query: query, page: 1, limit: 20)
+            case .instances:
+                instances = try await MlemStats.main.searchInstances(query: query)
+            }
+        } catch {
+            handleError(error)
+        }
+    }
```

Comment:
This whole thing is being run on the main actor from L80. I suggest removing that `@MainActor` annotation and adopting a similar construction to the search sheet suggestion (assuming `@MainActor` setters from previous comment):

```suggestion
    func refresh() async {
        do {
            if !query.isEmpty {
                try await Task.sleep(for: .seconds(0.2))
            }
            setCommunities(.init())
            setPeople(.init())
            setInstances(.init())
            switch selectedTab {
            case .communities:
                setCommunities(try await appState.firstApi.searchCommunities(query: query, page: 1, limit: 20))
            case .users:
                setPeople(try await appState.firstApi.searchPeople(query: query, page: 1, limit: 20))
            case .instances:
                setInstances(try await MlemStats.main.searchInstances(query: query))
            }
        } catch {
            handleError(error)
        }
    }
```